builder/content/geometry/geometry.hpp

```diff
@@ -384,3 +384,94 @@ vector<pair<Pt, Pt>> tangents(const Pt& o1,
 	}
 	return res;
 }
+// Returns the smallest enclosing circle of `v`
+pair<Pt, db> welzl(vector<Pt> v)
+{
+	int n = SZ(v), k = 0, idxes[2];
+	shuffle(ALL(v), rng);
+	Pt c = v[0];
+	db r = 0;
+	while (true)
+	{
+		FOR(i, k, n)
+		{
+			if (sgn(abs(v[i] - c) - r) > 0)
+			{
+				swap(v[i], v[k]);
+				if (k == 0)
+					c = v[0];
+				else if (k == 1)
+					c = (v[0] + v[1]) / 2;
+				else
+					c = circumCenter(
+						v[0], v[1],	v[2]);
+				r = abs(v[0] - c);
+				if (k < i)
+				{
+					if (k < 2)
+						idxes[k++] = i;
+					shuffle(v.begin() + k,
+						v.begin() + i + 1, rng);
+					break;
+				}
+			}
+			while (k > 0 && idxes[k - 1] == i)
+				k--;
+			if (i == n - 1)
+				return {c, r};
+		}
+	}
+}
+// Returns the half-plane intersection
+vector<Pt> hplaneInter(vector<Line> lines)
+{
+	const db INF = 1e9;
```

Comment:
Use different constant name to prevent confusion. INF is already reserved for a global INF value.